zulipterminal/model.py

```diff
@@ -555,6 +559,22 @@ def _stream_info_from_subscriptions(
                    key=lambda s: s[0].lower())
         )
 
+    def _group_info_from_realm_user_groups(self,
+                                           groups: List[Dict[str, Any]]
+                                           ) -> List[str]:
+        """
+        Stores group information in the index and returns a list of
+        group_names which helps in group typeahead. (Eg: @*terminal*)
+        """
+        for sub_group in groups:
+            self.index['groups'][sub_group['id']] = {
+                key: sub_group[key] for key in sub_group if key != 'id'}
```

Comment:
If we do keep all this data around, my feeling is that we could break this up:
* Setup the full dict
* Extract just the names from the dict, for use with matching

Reply:
We could have a further chat about this. I thought of going with just storing it in the model, based on your previous comments. As and when we need the full details, (Not foreseeable future at least) we could benefit from storing it in the index.